litellm/integrations/amberflo/amberflo_metering.py

```diff
@@ -0,0 +1,115 @@
+# What is this?
+## On Success events log usage to Amberflo
+
+import json
+import os
+from typing import Optional
+
+import httpx
+
+import litellm
+from litellm._logging import verbose_logger
+from litellm.integrations.custom_logger import CustomLogger
+from litellm.llms.custom_httpx.http_handler import (
+    HTTPHandler,
+    get_async_httpx_client,
+    httpxSpecialProvider,
+)
+
+class AmberfloLogger(CustomLogger):
+    def __init__(self) -> None:
+        super().__init__()
+        self.validate_environment()
+        self.async_http_handler = get_async_httpx_client(
+            llm_provider=httpxSpecialProvider.LoggingCallback
+        )
+        self.sync_http_handler = HTTPHandler()
+
+    def validate_environment(self):
+        """
+        Expects
+        AMBERFLO_API_KEY
+
+        in the environment
+        """
+        missing_keys = []
+        if os.getenv("AMBERFLO_API_KEY", None) is None:
+            missing_keys.append("AMBERFLO_API_KEY")
+
+        if len(missing_keys) > 0:
+            raise Exception("Missing keys={} in environment.".format(missing_keys))
+
+    def _common_logic(self, kwargs: dict, response_obj) -> dict:
+        id = response_obj.get("id", kwargs.get("litellm_call_id"))
+
+        dimensions = {
+            "model": response_obj.get("model", kwargs.get("model")),
+            "object": response_obj.get("object"),
+        }
+
+        created = response_obj.get("created", None)
+
+        user = response_obj.get("user", kwargs.get("user"))
+        if not user:
+            raise Exception("Amberflo: user is required")
+
+        usage = {}
+        if isinstance(response_obj, (litellm.ModelResponse, litellm.EmbeddingResponse)) and hasattr(response_obj, "usage"):
+            usage = {
+                "prompt_tokens": response_obj["usage"].get("prompt_tokens", 0),
+                "completion_tokens": response_obj["usage"].get("completion_tokens", 0),
+            }
+
+        return {
+            "customerId": user,
+            "values": {"input_tokens": usage.get("prompt_tokens"), "output_tokens": usage.get("completion_tokens")},
+            "meterTimeInMillis": created * 1000,
+            "dimensions": dimensions,
+            "uniqueId": id,
+        }
```

Comment:
in the case the `if` part does not execute, then you'll have an invalid ingest message.

move this map to inside the if and return `None` outside.